ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Workbenches.resource

```diff
@@ -321,15 +321,23 @@ Start Workbench
 
 Access To Workbench
     [Documentation]    Gets access to a workbench from DS Project details page after it got launched
-    [Arguments]    ${username}  ${password}  ${auth_type}
+    [Arguments]    ${username}  ${password}  ${auth_type}    ${expected_ide}=JupyterHub
     Run Keyword And Warn On Failure    Wait Until Page Contains    Log in with OpenShift    timeout=15s
     ${oauth_prompt_visible} =    Is OpenShift OAuth Login Prompt Visible
     IF  ${oauth_prompt_visible}    Click Button     Log in with OpenShift
     Run Keyword And Warn On Failure   Login To Openshift  ${username}  ${password}  ${auth_type}
     ${authorization_required}=  Is Service Account Authorization Required
     IF  ${authorization_required}  Authorize Jupyterhub Service Account
-    Wait Until Page Contains Element  xpath://div[@id="jp-top-panel"]  timeout=60s
-    Maybe Close Popup
+    IF  "${expected_ide}"=="VSCode"
+        Wait Until Page Contains Element  xpath://div[@class="menubar-menu-button"]  timeout=60s
+        Wait Until Page Contains Element  xpath://div[@class="monaco-dialog-box"]  timeout=60s
+        Wait Until Page Contains  Do you trust the authors of the files in this folder?
+    ELSE IF  "${expected_ide}"=="JupyterHub"
+        Wait Until Page Contains Element  xpath://div[@id="jp-top-panel"]  timeout=60s
+        Maybe Close Popup
+    ELSE
+        Fail    msg=Unknown IDE typy given: '${expected_ide}'. Please check and fix or implement.
+    END
```

Comment:
why not re-using `Wait Notebook To Be Loaded`?

Reply:
Yeah, I was thinking about it. Unfortunately, I would have to know the ID of the image here. Since there is a different approach how the workbench is spawned via JupyterHub vs Data Science project, I don't know the image ID that was spawned. If I decide to propagate image ID instead of the expected IDE variable, it means more changes on more places. With this approach I just put `JupyterLab` as a default, since it was only one other value before.

Another thing is that in JupyterHub launcher, there are different IDs for images used than what is used to determine in DS projects - but (even though not nice) that would be easy to fix with amending the `Notebook Expected Ide` so this argument doesn't apply fully.

---

Let me think about it more, maybe I'll amend.

Reply:
I think I would keep this as is for now not mixing the approaches for JupyterLab and DS projects at the moment. Let me know in case you're strongly against it. Thanks!